src/serialmonitor/serialportctrl.ts

```diff
@@ -102,7 +105,7 @@ export class SerialPortCtrl {
         this._child.stdout.on("data", (data) => {
             if (this.isActive) {
                 const jsonObj = JSON.parse(data.toString())
-                this._bufferedOutputChannel.append(jsonObj["payload"] + "\n");
+                this._bufferedOutputChannel.append(jsonObj["timestamp"] + jsonObj["payload"] + "\n");
```

Comment:
One thing to consider here is forcing a certain "break" rather than allowing the user to set it. If there arises a scenario where the user wants to do something with the timestamp data, they might have trouble doing that along with formatting, since they'd have to parse out the extra characters.

i.e. Hard-code in the "->" and only let the user define the format of the time. This way they only get the time in the 'timestamp' json and they don't have to parse out the seperator characters. This would mean users only set '%H:%m:%s' for example, compared to now, where they can set '%H:%m:%s ->' 

However, for how the extension currently uses it, what you have is fine and this comment won't block your PR.